packages/request-client.js/src/api/request.ts

```diff
@@ -574,6 +624,10 @@ export default class Request {
    * @returns The updated request data
    */
   public getData(): Types.IRequestDataWithEvents {
+    if (this.confirmationErrorOccurred) {
+      throw Error('request confirmation failed');
```

Comment:
The message should be more explicit
We could define `confirmationErrorOccurred` as a string and insert the error occurred inside to print it here

Reply:
We get the error from an event already. I think it's enough.